### PR TITLE
🤖 refactor: remove workflow actions

### PR DESCRIPTION
**Pull request by ThomasK33**
## Summary

Removes the workflow action subsystem from durable workflows and restores the deleted deep review, security scan, and simplify workflows as pure sub-agent conductors. The restored workflows now delegate Git/security/environment inspection to structured-output sub-agents and keep patching behind the existing workflow patch boundary.

## Background

Workflow actions had become a second orchestration path alongside workflow-owned sub-agents. This change collapses workflow orchestration onto the sub-agent model while preserving legacy workflow run parsing for historical action events.

## Implementation

- Removed action discovery/execution infrastructure, generated action content, built-in Git/GitHub action implementations, action child runtime, action-list tooling, and action-specific UI rendering.
- Simplified `WorkflowRunner`, `WorkflowService`, tool schemas, CLI/router wiring, docs generation, Docker build inputs, and workflow runtime sources around `agent`, `parallelAgents`, and `applyPatch`.
- Restored `deep-review-workflow`, `security-scan`, and project `simplify` as actionless workflows whose former action-shaped operations are now explicit sub-agent steps with `structuredOutput` schemas.
- Added workflow-agent `isolation: "none"` support and use it for read-only Git/security context collection so sub-agents can inspect the parent checkout, including staged/unstaged changes, without forking away the review target.
- Ran the `simplify` workflow on this branch and applied its follow-up simplifications: removed redundant workflow runtime fallback boilerplate and made `WorkflowService.listRuns` filter status snapshots before loading full run records.
- Updated the PR unit-test workflow filters so CI no longer tries to run deleted action-backed deep-review/security fixture names from `builtInWorkflowDefinitions.test.ts`, and isolated the new QuickJS workflow fixture from coverage.

## Validation

- `bun scripts/gen_builtin_workflows.ts check`
- `./scripts/generate-builtin-skills.sh`
- `bun test src/node/services/workflows/WorkflowTaskServiceAdapter.test.ts`
- `bun test src/node/services/workflows/subAgentWorkflowDefinitions.test.ts src/node/services/workflows/WorkflowDefinitionStore.test.ts src/node/services/workflows/builtInWorkflowDefinitions.test.ts`
- `bun test --max-concurrency=1 src/node/services/workflows/builtInWorkflowDefinitions.test.ts -t "built-in deep-research workflow"`
- `bun test --max-concurrency=1 src/node/services/workflows/subAgentWorkflowDefinitions.test.ts`
- `MUX_ESLINT_CONCURRENCY=1 make static-check`
- `/workflow simplify` on the branch, with its fixer validation including targeted workflow tests, `make typecheck`, and `MUX_ESLINT_CONCURRENCY=1 make lint`

## Risks

This is a broad workflow-runtime refactor. The main risks are regressions for existing persisted workflow runs and users with project/global action workflows. Legacy action events remain parseable, but workflow action execution/discovery APIs are intentionally removed. The restored built-in workflows are covered with targeted runtime tests proving structured sub-agent orchestration.

## Pains

A local Docker builder validation could not be completed because the Docker daemon is unavailable in this workspace; CI's Docker-related checks remain the source of truth for the Dockerfile COPY change.

---

_Generated with `mux` • Model: `openai:gpt-5.5` • Thinking: `xhigh` • Cost: `$72.86`_

### CODE REVIEW

**Comment by ThomasK33** [comment]
@codex review

**Comment by ThomasK33** [comment]
@codex review

Pushed a CI filter fix after the first unit failure: the PR unit job no longer asks Bun to run deleted deep-review/security fixture names from `builtInWorkflowDefinitions.test.ts`.

**Comment by ThomasK33** [comment]
Addressed Codex findings:

- `PRRT_kwDOPxxmWM6KzNb-` — added workflow-agent `isolation: "none"` support and marked read-only Git/security context collection/preflight steps in deep-review, simplify, and security-scan so they inspect the parent checkout rather than a fork missing staged/unstaged changes.
- `PRRT_kwDOPxxmWM6KzNcA` — moved `subAgentWorkflowDefinitions.test.ts` into the isolated unit-test list and excluded it from the coverage sweep, matching the existing QuickJS-heavy fixture handling.

**Comment by ThomasK33** [comment]
@codex review

Please review the pushed fixes for shared read-only workflow context and QuickJS fixture CI isolation.

**Comment by ThomasK33** [comment]
Addressed Codex finding `PRRT_kwDOPxxmWM6KzZ80`: removed the deleted workflow action generator wrapper and TS generator from the Dockerfile COPY inputs. Local `make static-check` passes. I attempted a local Docker builder validation, but the Docker daemon is unavailable in this workspace (`Cannot connect to the Docker daemon`).

**Comment by ThomasK33** [comment]
@codex review

Please review the Dockerfile COPY fix plus the prior shared-context and CI-isolation fixes.

**Comment by ThomasK33** [comment]
Addressed latest Codex findings:

- `PRRT_kwDOPxxmWM6Kzhxz` — moved security state persistence after the optional fix pass so `--fix` preflight runs against the reviewed checkout and persisted state can include the fix result.
- `PRRT_kwDOPxxmWM6Kzhx3` — security auto-fix now requires verifier proof state `verified` or `static_evidence` and `safeToFix: true`; `--no-verify --fix` no longer selects candidates for fixing.
- `PRRT_kwDOPxxmWM6Kzhx5` — deep-review loop budgeting now counts fixed IDs only when the parent patch application succeeded with `status: "applied"`.

Validation after the changes:
- `bun scripts/gen_builtin_workflows.ts check`
- `bun test src/node/services/workflows/subAgentWorkflowDefinitions.test.ts src/node/services/workflows/WorkflowDefinitionStore.test.ts src/node/services/workflows/builtInWorkflowDefinitions.test.ts`
- `MUX_ESLINT_CONCURRENCY=1 make static-check`

**Comment by ThomasK33** [comment]
@codex review

Please review the latest fix-pass edge case fixes.

**Comment by ThomasK33** [comment]
Addressed latest deep-review Codex findings:

- `PRRT_kwDOPxxmWM6Kzv4r` — added deterministic auto-fix skip guards before preflight/fixer for explicit diffs, explicit file snapshots, non-local targets, or missing local Git HEAD snapshots.
- `PRRT_kwDOPxxmWM6Kzv4x` — loop mode now continues after consuming the remaining fix budget so it performs a final no-fix verification pass; fix budget accounting still only counts successfully applied parent patches.

Validation after the changes:
- `bun scripts/gen_builtin_workflows.ts check`
- `bun test src/node/services/workflows/subAgentWorkflowDefinitions.test.ts src/node/services/workflows/WorkflowDefinitionStore.test.ts src/node/services/workflows/builtInWorkflowDefinitions.test.ts`
- `MUX_ESLINT_CONCURRENCY=1 make static-check`

**Comment by ThomasK33** [comment]
@codex review

Please review the latest deep-review auto-fix guard and loop follow-up pass fixes.

**Comment by ThomasK33** [comment]
Addressed latest deep-review Codex findings:

- `PRRT_kwDOPxxmWM6Kz4aV` — expanded non-local target detection to block auto-fix for PR-like targets, GitHub pull URLs anywhere in the target text, and `#123` shorthand, in addition to URLs/Git SSH targets.
- `PRRT_kwDOPxxmWM6Kz4aa` — `selectFixableIssues` now treats an empty synthesized `fixCandidateIds` list as selecting no fixes unless the user explicitly supplies `fixIssueIds`.

Validation after the changes:
- `bun scripts/gen_builtin_workflows.ts check`
- `bun test src/node/services/workflows/subAgentWorkflowDefinitions.test.ts src/node/services/workflows/WorkflowDefinitionStore.test.ts src/node/services/workflows/builtInWorkflowDefinitions.test.ts`
- `MUX_ESLINT_CONCURRENCY=1 make static-check`

**Comment by ThomasK33** [comment]
@codex review

Please review the latest deep-review non-local target and fixCandidateIds selection fixes.

**Comment by ThomasK33** [comment]
@codex review

Pushed a unit-test stabilization for the parallel workflow map step ordering observed in CI.

**Comment by ThomasK33** [comment]
Addressed Codex finding `PRRT_kwDOPxxmWM6K0Amk`: deep-review arg normalization now ignores raw `input` as a review target after flag parsing, so flag-only invocations fall back to `current workspace changes` while positional parsed targets still work.

Validation after the change:
- `bun scripts/gen_builtin_workflows.ts check`
- `bun test src/node/services/workflows/subAgentWorkflowDefinitions.test.ts src/node/services/workflows/WorkflowDefinitionStore.test.ts src/node/services/workflows/builtInWorkflowDefinitions.test.ts`
- `bun test src/node/services/workflows/WorkflowService.test.ts`
- `MUX_ESLINT_CONCURRENCY=1 make static-check`

**Comment by ThomasK33** [comment]
@codex review

Please review the flag-only deep-review target fix.

**Comment by ThomasK33** [comment]
@codex review

Rebased on latest `origin/main`, resolved the conflict in `aiService.ts`, and re-ran workflow/static validation before force-pushing.

**Comment by ThomasK33** [comment]
Addressed Codex finding `PRRT_kwDOPxxmWM6K0Nt9`: deep-review arg normalization now parses legacy `input` text first and then overlays schema-normalized structured args, so structured fields like `fix`, `baseRef`, and `target` win over raw input flags while legacy text invocations still work.

Validation after the change:
- `bun scripts/gen_builtin_workflows.ts check`
- `bun test src/node/services/workflows/subAgentWorkflowDefinitions.test.ts src/node/services/workflows/WorkflowDefinitionStore.test.ts src/node/services/workflows/builtInWorkflowDefinitions.test.ts`
- `MUX_ESLINT_CONCURRENCY=1 make static-check`

**Comment by ThomasK33** [comment]
@codex review

Please review the structured-args precedence fix.

**Comment by ThomasK33** [comment]
Addressed latest Codex findings:

- `PRRT_kwDOPxxmWM6K0Wzr` — simplify now parses legacy input text first and overlays structured args so explicit `baseRef`, `headRef`, `fix`, etc. win over raw transport text.
- `PRRT_kwDOPxxmWM6K0Wzz` — security-scan now uses the same precedence, and no longer treats raw `input` as the target after parsing flags.

Validation after the changes:
- `bun scripts/gen_builtin_workflows.ts check`
- `bun test src/node/services/workflows/subAgentWorkflowDefinitions.test.ts src/node/services/workflows/WorkflowDefinitionStore.test.ts src/node/services/workflows/builtInWorkflowDefinitions.test.ts`
- `MUX_ESLINT_CONCURRENCY=1 make static-check`

**Comment by ThomasK33** [comment]
@codex review

Please review the simplify/security structured-args precedence fixes.

**Comment by ThomasK33** [comment]
Addressed Codex finding `PRRT_kwDOPxxmWM6K0kvb`: security-scan metadata now declares `--finding` and `--finding-id` aliases so slash/CLI text does not fail normalization before the workflow-local repeated-ID parser runs. Structured `finding`/`findingId` values are also merged into `fixFindingIds`.

Validation after the change:
- `bun scripts/gen_builtin_workflows.ts check`
- `bun test src/node/services/workflows/subAgentWorkflowDefinitions.test.ts src/node/services/workflows/WorkflowDefinitionStore.test.ts src/node/services/workflows/builtInWorkflowDefinitions.test.ts`
- `MUX_ESLINT_CONCURRENCY=1 make static-check`

**Comment by ThomasK33** [comment]
@codex review

Please review the security targeted-fix flag schema fix.

**Comment by ThomasK33** [comment]
Addressed Codex finding `PRRT_kwDOPxxmWM6K0kvi`: deep-review now falls back to the reviewed `gitContext.status.headSha` for `expectedHeadSha` and skips fix application if no reviewed local HEAD is available. I also applied the same reviewed-HEAD fence pattern to simplify and security-scan fix application paths.

Validation after the change:
- `bun scripts/gen_builtin_workflows.ts check`
- `bun test src/node/services/workflows/subAgentWorkflowDefinitions.test.ts src/node/services/workflows/WorkflowDefinitionStore.test.ts src/node/services/workflows/builtInWorkflowDefinitions.test.ts`
- `MUX_ESLINT_CONCURRENCY=1 make static-check`

**Comment by ThomasK33** [comment]
@codex review

Please review the reviewed-HEAD fence fix.

**Comment by ThomasK33** [comment]
@codex review

Addressed the reviewed HEAD fence findings:
- deep-review and simplify now use the reviewed Git snapshot as the patch fence and skip when preflight reports a different HEAD.
- security-scan persistence and auto-fixes now pass the scan HEAD to patch application, skipping persistence/fixes when that reviewed HEAD is unavailable.
- added isolated workflow coverage for the deep-review mismatch path and security persistence expectedHeadSha.

**Comment by ThomasK33** [comment]
@codex review

Addressed the security persistence follow-up: fixed security scans now bundle `.mux/security` persistence into the same fixer child patch and apply exactly one reviewed patch at the scan HEAD. Read-only/skipped/no-op fix runs still use the normal `security-write-state` persistence path fenced to the reviewed scan HEAD. Added isolated workflow coverage asserting the fixed path only applies `apply-security-fixes` with the reviewed HEAD.

**Comment by ThomasK33** [comment]
@codex review

Addressed the failed-fix fallback: if the bundled `apply-security-fixes` patch fails or conflicts, the workflow now falls through to the state-only `security-write-state` persistence path fenced to the reviewed scan HEAD. Successful fix patches still return immediately with persistence bundled into that single reviewed patch.

**Comment by ThomasK33** [comment]
@codex review

Addressed verifier ID validation: deep-review now only attaches verifier results when `issueId` matches the candidate issue, otherwise it downgrades to `unclear`; security-scan now only attaches verifier results when `findingId` matches the candidate finding, otherwise it marks the finding as `needs_human_review` and `safeToFix: false`. This prevents mismatched verifier outputs from selecting the wrong item for auto-fix.

**Comment by ThomasK33** [comment]
@codex review

Addressed dirty scan-target persistence: the state-only `apply-security-state` path now passes `force: true` while retaining the reviewed `expectedHeadSha` fence. This is limited to the `.mux/security` persistence patch path; code-fix patches remain non-forced.

**Comment by ThomasK33** [comment]
@codex review

Please re-check after the latest `force: true` state-persistence patch. All previously reported threads are resolved and CI is green.

**Comment by ThomasK33** [comment]
@codex review

Addressed the simplify cap: the workflow now builds a capped `fixSynthesis` from `actionableFindings.slice(0, maxFindings)` and uses that capped payload for preflight skips, fixer prompts, and fix-mode structured output. The fixer no longer receives findings beyond `--max-findings`.

**Comment by ThomasK33** [comment]
@codex review

Addressed the simplify regressions:
- untracked-only current changes now skip with explicit `git add -N`/stage guidance unless diff content is present, avoiding review of filenames without file contents;
- simplify text args again accept `--base=...`, `--trunk=...`, `--head=...`, and `--max-findings=...` equals-form flags.

**Comment by ThomasK33** [comment]
@codex review

Rebased `workflows-7pag` onto `origin/main` and re-ran local validation:
- `bun test src/node/services/workflows/subAgentWorkflowDefinitions.test.ts`
- `make static-check`

File: src/node/services/workflows/WorkflowService.ts [hunk 1]
```diff
@@ -333,7 +300,6 @@ export class WorkflowService {
       );
       settleStatusWrite();
       await this.notifyRunStatusChanged(interrupted);
-      await this.interruptChildWorkflowRuns(input.runId, input.workspaceId, new Set([input.runId]));
       await (this.taskAdapterFactory?.(input.runId) ?? this.requireTaskAdapter()).interruptRun?.();
```
[the remark this answers is not in the document]
Comment (ThomasK33) [reply src/node/services/workflows/WorkflowService.ts:318]: Fixed in 9bea4ce44 by restoring recursive cleanup for persisted child workflow runs during parent interruption. `interruptRun` now walks non-terminal descendant runs linked by `parentWorkflow`, interrupts each run-scoped adapter, preserves terminal children, and includes a regression test covering parent/child/grandchild cleanup before return.

**Comment by ThomasK33** [comment]
@codex review

Addressed the nested child workflow interruption feedback and pushed `9bea4ce44`.

Local validation:
- `bun test src/node/services/workflows/WorkflowService.test.ts`
- `MUX_ESLINT_CONCURRENCY=1 make static-check`

File: src/node/builtinWorkflows/deep-review-workflow.js [hunk 2]
```diff
@@ -1,1913 +1,874 @@
...
-  baseFix.selectedIssues = selected.map(function (item) {
-    return summarizeFixIssue(item.issueId, item.issue);
-  });
-  context.log("Selected deep review fixes", {
-    selectedCount: selected.length,
-    skippedCount: context.candidates.length - selected.length,
+  const preflight = collectGitPreflightAgent(context.agent, input, gitContext, suffix);
+  if (!preflight.ok) {
+    return appendFixSkipped(result, preflight.reason || "Auto-fix preflight failed.", preflight);
+  }
```
[the remark this answers is not in the document]
Comment (ThomasK33) [reply src/node/builtinWorkflows/deep-review-workflow.js:263]: Fixed in 36ef9a9ea. The deep-review, security-scan, and simplify fix flows now compare the preflight agent’s current `headSha` against the reviewed snapshot before spawning any fixer, while keeping the apply-time expected HEAD fence for movement after preflight. I also updated the deep-review regression test so a spoofed preflight with `headSha` moved but `expectedHeadSha` still equal to the reviewed SHA skips before `fix-review-findings` runs.

**Comment by ThomasK33** [comment]
@codex review

Addressed the preflight current-HEAD feedback and pushed `36ef9a9ea`.

Local validation:
- `bun test src/node/services/workflows/subAgentWorkflowDefinitions.test.ts`
- `bun test src/node/services/workflows/WorkflowService.test.ts`
- `MUX_ESLINT_CONCURRENCY=1 make static-check`

File: src/node/builtinWorkflows/security-scan.js [hunk 3]
```diff
@@ -1,1439 +1,960 @@
...
+    const persistence = attemptedFixPatch
+      ? await persistSecurityStateAgent(agent, persistencePayload)
+      : persistSecurityStateAgent(agent, persistencePayload);
+    applyResult = await mux.patch.applySafely({
+      id: "apply-security-state",
+      source: persistence,
+      expectedHeadSha: persistenceExpectedHeadSha,
+      // Security state patches are constrained by prompt to .mux/security and
+      // should persist even when the scan target includes uncommitted work.
+      force: true,
```
[the remark this answers is not in the document]
Comment (ThomasK33) [reply src/node/builtinWorkflows/security-scan.js]: Fixed in ce86ea4ed. `apply-security-state` now passes `allowedPathPrefixes: [".mux/security"]`, and workflow patch application validates the actual git-format-patch mbox paths before dry-run/real apply. A forced security-state patch that touches source files is rejected before `applyPatchArtifact` runs, covered by `WorkflowTaskServiceAdapter` regression coverage.

File: src/node/builtinWorkflows/deep-review-workflow.js [hunk 4]
```diff
@@ -1,1913 +1,878 @@
...
+  const fixer = context.agent({
+    id: stepId("fix-review-findings", suffix),
+    title: "Fix verified review findings",
+    agentId: EXEC_AGENT_ID,
+    prompt: fixPrompt(reviewInput, fixableIssues, preflight),
+    outputSchema: fixerSchema(),
+  });
+  const fixerOutput = fixer.structuredOutput || { madeChanges: false, fixedIssueIds: [] };
+  if (!fixerOutput.madeChanges)
+    return appendFixNoChanges(result, fixer.reportMarkdown, fixerOutput);
```
[the remark this answers is not in the document]
Comment (ThomasK33) [reply src/node/builtinWorkflows/deep-review-workflow.js:284]: Fixed in ce86ea4ed. Deep-review now requires the fixer output `fixedIssueIds` to intersect the selected fixable issue IDs before applying the patch; otherwise it records a skipped fix with the fixer output and leaves the parent checkout untouched.

File: .mux/workflows/simplify.js [hunk 5]
```diff
@@ -278,14 +188,10 @@ export default async function simplifyWorkflow({ args, phase, log, action, agent
...
+    prompt: fixPrompt(reviewContext, fixSynthesis, preflight),
+    outputSchema: fixerSchema(),
   });
-  const fixerOutput = mux.utils.mustObject(
-    fixer.structuredOutput,
-    "fixer structured output is required"
-  );
-
+  const fixerOutput = fixer.structuredOutput;
   if (!fixerOutput.madeChanges) {
```
[the remark this answers is not in the document]
Comment (ThomasK33) [reply .mux/workflows/simplify.js:195]: Fixed in ce86ea4ed. Simplify now requires `fixedFindingIds` to intersect the selected actionable finding IDs before calling `applySafely`; mislabelled or unrelated fixer commits are reported as skipped with no parent mutation.

**Comment by ThomasK33** [comment]
@codex review

Addressed the latest patch-safety feedback and pushed `ce86ea4ed`.

Local validation:
- `bun test src/node/services/workflows/WorkflowTaskServiceAdapter.test.ts`
- `bun test src/node/services/workflows/subAgentWorkflowDefinitions.test.ts`
- `bun test src/node/services/workflows/WorkflowRunner.test.ts -t "applyPatch"`
- `MUX_ESLINT_CONCURRENCY=1 make static-check`

File: src/node/services/workflows/WorkflowTaskServiceAdapter.ts [hunk 6]
```diff
@@ -206,6 +220,81 @@ export class WorkflowTaskServiceAdapter implements WorkflowTaskAdapter {
...
+    }
+
+    const projectArtifacts = artifact.projectArtifacts.filter(
+      (projectArtifact) =>
+        spec.projectPath == null || projectArtifact.projectPath === spec.projectPath
+    );
+    const violations = new Set<string>();
+    for (const projectArtifact of projectArtifacts) {
+      if (projectArtifact.status !== "ready") {
+        continue;
```
[the remark this answers is not in the document]
Comment (ThomasK33) [reply src/node/services/workflows/WorkflowTaskServiceAdapter.ts]: Fixed in 9f61d369b. The adapter now lets the dry-run step perform the existing pending-artifact wait, then validates the refreshed mbox paths before the real apply. If an allowlisted patch artifact is still non-ready at validation time, it fails closed instead of proceeding to apply. The regression now verifies only the dry-run call happens before an out-of-fence patch is rejected.

**Comment by ThomasK33** [comment]
@codex review

Addressed the pending-artifact path-validation feedback and pushed `9f61d369b`.

Local validation:
- `bun test src/node/services/workflows/WorkflowTaskServiceAdapter.test.ts`
- `bun test src/node/services/workflows/subAgentWorkflowDefinitions.test.ts`
- `bun test src/node/services/workflows/WorkflowRunner.test.ts -t "applyPatch"`
- `MUX_ESLINT_CONCURRENCY=1 make static-check`

File: src/node/services/workflows/WorkflowTaskServiceAdapter.ts [hunk 7]
```diff
@@ -206,6 +221,81 @@ export class WorkflowTaskServiceAdapter implements WorkflowTaskAdapter {
...
+    }
+
+    const projectArtifacts = artifact.projectArtifacts.filter(
+      (projectArtifact) =>
+        spec.projectPath == null || projectArtifact.projectPath === spec.projectPath
+    );
+    const violations = new Set<string>();
+    for (const projectArtifact of projectArtifacts) {
+      if (projectArtifact.status !== "ready") {
+        return `Patch artifact for ${projectArtifact.projectName} is ${projectArtifact.status}; cannot validate allowedPathPrefixes.`;
```
[the remark this answers is not in the document]
Comment (ThomasK33) [reply src/node/services/workflows/WorkflowTaskServiceAdapter.ts:255]: Fixed in 6d55dccf7. Path validation now treats `skipped` project artifacts as no-op siblings and continues validating ready artifacts; pending/failed artifacts still fail closed. The adapter regression includes a skipped sibling before the ready artifact to cover the multi-project case.

**Comment by ThomasK33** [comment]
@codex review

Addressed the skipped-artifact path-validation feedback and pushed `6d55dccf7`.

Local validation:
- `bun test src/node/services/workflows/WorkflowTaskServiceAdapter.test.ts`
- `MUX_ESLINT_CONCURRENCY=1 make static-check`

**Comment by ThomasK33** [comment]
@codex review

Rebased `workflows-7pag` onto latest `origin/main`, resolved conflicts, and force-pushed `e77025727`.

Local validation after the rebase:
- `bun test src/node/services/workflows/WorkflowRunner.test.ts -t "outputSchema"`
- `bun test src/node/services/workflows/WorkflowTaskServiceAdapter.test.ts`
- `bun test src/node/services/workflows/subAgentWorkflowDefinitions.test.ts`
- `bun test src/node/services/workflows/WorkflowService.test.ts -t "interrupts nested child workflow runs before returning"`
- `MUX_ESLINT_CONCURRENCY=1 make static-check`

**Comment by ThomasK33** [comment]
@codex review

Fixed the post-rebase WorkflowRunner unit failure and pushed `7dd1d4f5b`.

Additional local validation:
- `bun test src/node/services/workflows/WorkflowRunner.test.ts -t "retries parallelAgents validation failures before slower siblings finish"`
- `bun test src/node/services/workflows/WorkflowService.test.ts`
- `MUX_ESLINT_CONCURRENCY=1 make static-check`

File: src/node/services/workflows/WorkflowRunner.ts [hunk 8]
```diff
@@ -2092,16 +1257,12 @@ export class WorkflowRunner {
...
-          resultSpec: getWorkflowAgentResumedResultSpec(step.spec, {
-            allowLegacyInvalidOutputSchema: step.allowMissingOutputSchema,
-          }),
-        };
+        const rawResult = await this.taskAdapter.waitForAgentTask(
+          step.taskId,
+          omitWorkflowAgentOutputSchema(step.spec),
+          step.waitOptions
+        );
+        return { rawResult, resultSpec: omitWorkflowAgentOutputSchema(step.spec) };
```
[the remark this answers is not in the document]
Comment (ThomasK33) [reply src/node/services/workflows/WorkflowRunner.ts]: Fixed in 7dd1d4f5b. Existing/bulk-created tasks now preserve their valid outputSchema for result validation; only legacy runs with `agentOutputSchemaRequired: false` omit invalid/missing schemas. The previously failing parallelAgents validation retry test passes locally.

File: src/node/services/workflows/WorkflowTaskServiceAdapter.ts [hunk 9]
```diff
@@ -205,6 +221,84 @@ export class WorkflowTaskServiceAdapter implements WorkflowTaskAdapter {
...
+  ): Promise<string | undefined> {
+    if (spec.allowedPathPrefixes == null || spec.allowedPathPrefixes.length === 0) {
+      return undefined;
+    }
+    const workspaceSessionDir = this.patchToolConfig?.workspaceSessionDir;
+    if (workspaceSessionDir == null || workspaceSessionDir.length === 0) {
+      return "applyPatch allowedPathPrefixes requires patch artifact metadata";
+    }
+
+    const artifact = await readSubagentGitPatchArtifact(workspaceSessionDir, spec.sourceTaskId);
```
[the remark this answers is not in the document]
Comment (ThomasK33) [reply src/node/services/workflows/WorkflowTaskServiceAdapter.ts]: Fixed in e3464b1ea. Path-fenced apply validation now reuses the same workspace-or-ancestor artifact lookup as the dry-run apply before reading the mbox, with direct current-session lookup retained as a fallback for tests/configs without a workspace id.

**Comment by ThomasK33** [comment]
@codex review

Addressed the latest WorkflowRunner output-schema and ancestor artifact lookup feedback and pushed `e3464b1ea`.

Local validation:
- `bun test src/node/services/workflows/WorkflowRunner.test.ts -t "retries parallelAgents validation failures before slower siblings finish"`
- `bun test src/node/services/workflows/WorkflowTaskServiceAdapter.test.ts`
- `MUX_ESLINT_CONCURRENCY=1 make static-check`

**Comment by ThomasK33** [comment]
@codex review

Latest push fixes the remaining WorkflowService mock workflow fixtures so required agent output schemas are paired with structured mock outputs. Local validation: `bun test src/node/services/workflows/WorkflowService.test.ts` and `MUX_ESLINT_CONCURRENCY=1 make static-check`.

**Comment by ThomasK33** [comment]
@codex review

Follow-up: reproduced the failed `Test / Integration` job locally with `TEST_INTEGRATION=1 bun x jest tests/ui/chat/sendModeDropdown.test.ts --runInBand`, then increased the slow end-to-end-style send-mode case timeout for loaded CI. Validation also passed with `MUX_ESLINT_CONCURRENCY=1 make static-check`.
